editor/editor_help.cpp

```diff
@@ -2381,9 +2381,38 @@ static void _add_text_to_rt(const String &p_bbcode, RichTextLabel *p_rt, Control
 	bbcode = bbcode.replace("\n[/codeblock]", "[/codeblock]");
 	bbcode = bbcode.replace("[/codeblock]\n", "[/codeblock]");
 
+	// First pass, iterate through all lines.
+	// Find a list and its elements, replace with a BBcode list (ul).
+	int newline_pos = -1;
+	int inside_list = false;
+	while (true) {
+		newline_pos = bbcode.find_char('\n', newline_pos + 1);
+		if (newline_pos == -1) {
+			break;
+		}
+
+		if (!inside_list && bbcode[newline_pos + 1] == '-') {
+			bbcode = bbcode.left(newline_pos) + "[ul]" + bbcode.right(-newline_pos - 2);
```

Comment:
```suggestion
			bbcode = bbcode.left(newline_pos) + "[ul]" + bbcode.right(-(newline_pos + 2));
```
Far better readability IMO